File: mine/sorts/bubble.cpp
/*************************************************************************
    > File Name: bubble.cpp
    > Author: Feng
    > Created Time: Wednesday, December 05, 2018 PM04:53:12 CST
    > Content: 
 ************************************************************************/

#include <iostream>

using namespace std;

void bubble(int32_t array[], uint32_t array_len)
{
    for (uint32_t i = 0; i < array_len - 1; ++i) {
        for (uint32_t j = 0; j < array_len - 1 - i; ++j) {
            if (array[j] > array[j + 1]) {
                int32_t tmp = array[j];
                array[j] = array[j + 1];
                array[j + 1] = tmp;
            }
        }
    }
}

int main()
{
    int32_t array[11] = {105, -5, -89, 1, 9, 100, 3, -8, 99, 1, 89};
    bubble(array, 11);
    for (uint32_t i = 0; i < 11; ++i) {
        cout << array[i] << " "; 
    }
    cout << endl;

    return 0;
}

File: mine/sorts/bubble.py
#!/usr/bin/env python3
#-*- coding: utf-8 -*-
 
# File Name: bubble.py
# Author: Feng
# Created Time: Sunday, December 02, 2018 PM09:26:22 CST
# Content: 

def bubble(array):
    length = len(array)
    for i in range(length - 1 - 1):
        for j in range(length - 1 - i):
            if array[j] > array[j + 1]:
                array[j], array[j + 1] = array[j + 1], array[j]
    return array

if __name__ == '__main__':
    test_array = [100, -2, -5, 1, 200, 5, 3, 201]
    print(bubble(test_array))

